pkg/util/validation/limits.go

```diff
@@ -130,7 +130,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&l.EnforceMetricName, "validation.enforce-metric-name", true, "Enforce every sample has a metric name.")
 	f.BoolVar(&l.EnforceMetadataMetricName, "validation.enforce-metadata-metric-name", true, "Enforce every metadata has a metric name.")
 
-	f.IntVar(&l.MaxSeriesPerQuery, "ingester.max-series-per-query", 100000, "The maximum number of series for which a query can fetch samples from each ingester. This limit is enforced only in the ingesters (when querying samples not flushed to the storage yet) and it's a per-instance limit. This limit is ignored when running the Cortex blocks storage.")
+	f.IntVar(&l.MaxSeriesPerQuery, "ingester.max-series-per-query", 100000, "The maximum number of series for which a query can fetch samples from each ingester and block storage. This limit is enforced in the ingesters and block storage and it's a per-instance limit.")
```

Comment:
While writing this I just realized we're introducing a breaking change. I need to discuss it next week with Peter/Goutham, but we may need to introduce a new limit for the blocks storage to avoid introducing a breaking change in the logic.

```suggestion
	f.IntVar(&l.MaxSeriesPerQuery, "ingester.max-series-per-query", 100000, "The maximum number of series for which a query can fetch samples from ingesters and storage. When running the chunks storage, this limit is enforced only in the ingesters and it's a per-instance limit. When running the blocks storage, this limit is enforced in the querier on samples read both from ingesters and storage and it's a per-query limit.")
```